app/src/main/java/com/bpn/trainme/presentation/features/home/HomeScreen.kt

```diff
@@ -29,8 +34,15 @@ fun HomeScreen(
     onNavigateToSettings: () -> Unit
 ){
     val viewModel = hiltViewModel<HomeViewModel>()
+
+    val scope = rememberCoroutineScope()
     val uiState = viewModel.uiState.collectAsState()
```

Comment:
you may use `collectAsStateWithLifecycle()` to make the collection of the state lifecycle aware